### ISSUE TITLE
SwiftUIFlux, ConnectedView, @EnvironmentObject: "A View.environmentObject(_:) for Store<AppState> may be missing as an ancestor of this view"

### ISSUE THREAD

**Issue by d4rkd3v1l** [nalexn/ViewInspector#93]
Did anyone ever try to test [ConnectedView](https://github.com/Dimillian/SwiftUIFlux/blob/master/Sources/SwiftUIFlux/connector/ConnectedView.swift)s from [SwiftUIFlux](https://github.com/Dimillian/SwiftUIFlux) with `ViewInspector`?

I always end up with following error in [StoreConnector](https://github.com/Dimillian/SwiftUIFlux/blob/39b885558fdee277f7ca2a8c81900123e9bc436f/Sources/SwiftUIFlux/connector/StoreConnector.swift#L15), line 15, where the `store` is accessed.
> Thread 1: Fatal error: No ObservableObject of type Store<AppState> found. A View.environmentObject(_:) for Store<AppState> may be missing as an ancestor of this view.

<img width="802" alt="Screenshot 2021-03-13 at 11 28 55" src="https://user-images.githubusercontent.com/13336238/111027606-8bc17300-83f1-11eb-88e0-fce6b43c736c.png">

I already tried those async method described here, without success. I feel it doesn't work because of wrapping the [StoreConnector](https://github.com/Dimillian/SwiftUIFlux/blob/master/Sources/SwiftUIFlux/connector/StoreConnector.swift) in the extension of [ConnectedView](https://github.com/Dimillian/SwiftUIFlux/blob/39b885558fdee277f7ca2a8c81900123e9bc436f/Sources/SwiftUIFlux/connector/ConnectedView.swift#L26). But not quite sure about this, but somehow the passing of the `store` as `@EnvironmentObject` must be interrupted somewhere, I guess?

```
func testBla() throws {
    var sut = TimerView() // `ConnectedView`
    
    let exp = sut.on(\.didAppear) { view in // Error: view(TimerView.self).anyView() found SwiftUIFlux.StoreConnector instead of AnyView
        try view.anyView()
    }
    ViewHosting.host(view: sut.environmentObject(testStore))
    wait(for: [exp], timeout: 0.5)
}
```

I have to admit, I don't quite understand how `anyView()` can/should be used, from the name, I suggested, it can be "any view", but seems like this is not a correct assumption considering the error I get...
Anyway, when I try to provide the actual view like follows, I get the error mentioned above.

```
func testBla() throws {
    var sut = TimerView() // `ConnectedView`
    
    let exp = sut.on(\.didAppear) { view in
        try view.view(StoreConnector<AppState, TimerView>.self).find(button: "Start")
    }
    ViewHosting.host(view: sut.environmentObject(testStore))
    wait(for: [exp], timeout: 0.5)
}
```

I also tried the other more advanced `Inspection` async approach, but also no luck there, guess the problem is just how SwiftUIFlux is structured...

Any clues or help is highly appreciated. Thx in advance :-)

**Comment by nalexn** [comment]
Hey, yep, this is a known issue, and the fix is about to roll out.

**Comment by nalexn** [comment]
I've just released v0.7.0 which fixes the "No ObservableObject of type found" error. I see you have something else here.

There is a `StoreConnector` view from `SwiftUIFlux` that stands on your way to unwrap the inner hierarchy.
You'd need to add this extension in the test scope:

```swift
extension SwiftUIFlux.StoreConnector: Inspectable { }
```

Also, you don't need to explicitly unwrap `.view(StoreConnector<AppState, TimerView>.self)`, simply do `try view.find(button: "Start")`

**Comment by d4rkd3v1l** [comment]
Thx for your quick follow up.

I updated to 0.7.0 and applied your suggested changes -> still same error.

As far as I understand the `ViewInspector` code, the injection of the `EnvironmentObject` should happen in [ModifiedContent](https://github.com/nalexn/ViewInspector/blob/29771131e40b009232afc8d94d626d73081817bf/Sources/ViewInspector/SwiftUI/ModifiedContent.swift#L81) which should be called from the [Inspector](https://github.com/nalexn/ViewInspector/blob/29771131e40b009232afc8d94d626d73081817bf/Sources/ViewInspector/Inspector.swift#L177).

Unfortunately it seems like the `StoreConnector` is still in my way here.
<img width="1237" alt="Screenshot 2021-03-14 at 08 17 53" src="https://user-images.githubusercontent.com/13336238/111060531-d13b7a00-849d-11eb-8d7c-d032040c3a61.png">

In SwiftUIFlux you normally would inject the store via a `StoreProvider`:
```
StoreProvider(store: store) {
    ContentView()
}
```

But I also used it like follows e.g. for SwiftUI previews, which also works:
```
ContentView()
    .environmentObject(previewStore)
```

So could it be a problem to basically use the 2nd approach for my view tests, and maybe it should be more like the first StoreProvider one?

As I have my project on GitHub, you can just take a look at the code [here](https://github.com/d4rkd3v1l/timeARC/blob/ViewInspector/timeARCTests/TimerViewTests.swift) as well, if it helps.

**Comment by nalexn** [comment]
Ok, I forgot to add something in the 0.7.0, thanks to you now it's really fixed in 0.7.1. I did check that with your repo, and the test passes (although you need to change the searched text from "Start" to "start" - a typo)

**Comment by d4rkd3v1l** [comment]
Awesome 🤩 Thx, man!! Great support! 😀

Works like a charm... So guess it's time to write a ton of view tests now 😊

*typo was due to using `LocalizedStringKey`s which seem not to be resolved in the view, but that's just fine for me. I think actually `String`s automatically bridge to `LocalizedStringKey`s in SwiftUI normally, but as said, it's alright :-)
